src/components/form/VariableFieldRenderer.tsx

```diff
@@ -66,27 +63,57 @@ class VariableFormFieldRenderer extends PureComponent<Props> {
                 };
 
                 if (item[name]) {
-                    arr.push(obj);
+                    valueList.push(obj);
                 }
-
             }
         }
 
-        return arr;
-    }
-
-    constructor () {
-        super();
-
-        this.onAddHandler = this.onAddHandler.bind(this);
-        this.renderSubFields = this.renderSubFields.bind(this);
-        this.renderComponentSelector = this.renderComponentSelector.bind(this);
-        this.onChangeSubForm = this.onChangeSubForm.bind(this);
-        this.onChangeHeadHandler = this.onChangeHeadHandler.bind(this);
-        this.onChangeHandler = this.onChangeHandler.bind(this);
+        return (
+            <StyledFieldset>
+                {componentInfo && (
+                    <StyledDivider>
+                        <Label htmlFor={name}>Component</Label>
+                        <Select
+                            name={'Component'}
+                            options={SELECT_COMPONENT_TYPES}
+                            selected={name}
+                            onChange={e => this.onChangeHeadHandler(e)}
+                        />
+                    </StyledDivider>
+                )}
+                {valueList.map((valueItem: ValueItem, idx: number) =>
+                    valueItem && valueItem.value && (
+                        <StyledDivider key={idx}>
+                            <Label htmlFor={`${componentInfo.name}_${valueItem.name}_0`}>{valueItem.name}</Label>
+                            <KeyValue
+                                valueItem={valueItem}
+                                componentInfo={componentInfo}
+                                onSelectChangeHandler={this.onSelectChangeHandler}
+                                collection={this.props.match && this.props.match.params.collection}
+                            />
+                            <StyledInputWrapper>
+                                <StyledInput
```

Comment:
Er is niet per se altijd een invoerveld toch?

Reply:
goed gespot. vergeten de check ook hier weer toe te voegen